client/api.go

```diff
@@ -268,3 +268,89 @@ func (client Client) CallGetSingleRawSecretByNameV3(request GetSingleSecretByNam
 
 	return secretsResponse, nil
 }
+
+func (client Client) CallCreateProject(request CreateProjectRequest) (CreateProjectResponse, error) {
+
+	if request.Slug == "" {
+		request = CreateProjectRequest{
+			ProjectName:    request.ProjectName,
+			OrganizationId: request.OrganizationId,
+		}
+	}
+
+	var projectResponse CreateProjectResponse
+	response, err := client.Config.HttpClient.
+		R().
+		SetResult(&projectResponse).
+		SetHeader("User-Agent", USER_AGENT).
+		SetBody(request).
+		Post("api/v2/workspace")
+
+	if err != nil {
+		return CreateProjectResponse{}, fmt.Errorf("CallCreateProject: Unable to complete api request [err=%s]", err)
+	}
+
+	if response.IsError() {
+		return CreateProjectResponse{}, fmt.Errorf("CallCreateProject: Unsuccessful response. [response=%s]", response)
```

Comment:
can we make error logs format this way please? It would provide more details for users when they get error https://github.com/Infisical/infisical/blob/main/cli/packages/api/api.go#L29

Reply:
Slack thread 🧵

Marking as resolved as discussed.